lib/png.dx

```diff
@@ -93,11 +93,12 @@ def base64Decode (s:String) : Maybe String =
 
 Html : Type = List Char
 
-def makePNG (img:n=>m=>(Fin 3)=>Word8) : List Byte =
+def makePNG (img:n=>m=>(Fin 3)=>Word8) : List Byte = unsafeIO \().
   (AsList _ imgFlat) = toList for (i,j,k). img.i.j.k
-  (n, ptr) = (%ffi encodePNG (Int & CharPtr) (%getPtr imgFlat)
-               (size m) (size n))
-  AsList n $ for i. %ptrLoad (%ptrOffset ptr (ordinal i))
+  withTabPtr imgFlat \ptr.
+    (MkPtr rawPtr) = ptr
+    (n, ptr') = %ffi encodePNG (Int & CharPtr) rawPtr (size m) (size n)
+    AsList n $ tabFromPtr $ MkPtr ptr'
```

Comment:
The extra copies make me a little sad, but the fact that we no longer leak memory is sweet!

Reply:
I wonder where were memory leaks previously occurring? That is a bit surprising.